src/anet.c

```diff
@@ -420,11 +434,19 @@ static int anetTcpGenericConnect(char *err, const char *addr, int port,
         /* Try to create the socket and to connect it.
          * If we fail in the socket() call, or on connect(), we retry with
          * the next entry in servinfo. */
-        if ((s = socket(p->ai_family,p->ai_socktype,p->ai_protocol)) == -1)
+        int sockflags = 0;
+#ifdef SOCK_NONBLOCK
+        if (flags & ANET_CONNECT_NONBLOCK) p->ai_socktype |= SOCK_NONBLOCK;
+#else
+        if (flags & ANET_CONNECT_NONBLOCK) sockflags |= O_NONBLOCK;
+#endif
```

Comment:
i'm confused.
why do we have to pass two different flags for the same purpose, rather than let anetCreateSocket handle it implicitly with ifdefs?
as far as i remember that was your previous approach, why did you abandon it?

Reply:
> since we do care about CLOEXEC for the listen sockets, maybe we should handle them like NONBLOCK REUSEADDR, and let anetCreateSocket get flags and handle them all internally?

Did I misunderstand your previous comment?

Reply:
Please check out the latest commit.

Reply:
@oranagra Any thoughts about the latest commit?

Reply:
Ping @oranagra

Reply:
Ping @oranagra